src/libgit2/CMakeLists.txt

```diff
@@ -1,9 +1,18 @@
 # libgit2: the shared library: this CMakeLists.txt compiles the core
 # git library functionality.
 
-add_library(libgit2 OBJECT)
-set_target_properties(libgit2 PROPERTIES C_STANDARD 90)
-set_target_properties(libgit2 PROPERTIES C_EXTENSIONS OFF)
+file(GLOB SRC_GIT2 *.c *.h
+	streams/*.c streams/*.h
+	transports/*.c transports/*.h)
+list(SORT SRC_GIT2)
+
+add_library(libgit2 STATIC ${SRC_GIT2})
```

Comment:
This was a bit confusing when we had an object library called `libgit2`, but (I think) it was only referenced as ` "$<TARGET_OBJECTS:libgit2>"` so it wasn't so bad... now (I think) we are doing things like `target_link_libraries(... libgit2)`. But that libgit2 refers (I think) to _this_ libgit2, not the actual... libgit2. 🫤 

Also, this produces a `liblibgit2.a` on disk which is not very attractive. Even if it is an intermediate step that consumers don't see, I think that we can improve this.

`git2common` or something? idk, maybe you have a better idea here.

Reply:
If I understand correclty a shared library canot be used for some reason. If so if user wants a shared library libgit2.so we could compile shared and static (with name libgit2_static) and link specificly wiith the later and install the former. If user wants only static then libgit2.a could be used for both uses. I guess this could be solve in an other PR ?